File: 2358-number-of-ways-to-split-array/number-of-ways-to-split-array.py
class Solution:
    def waysToSplitArray(self, nums: List[int]) -> int:
        rsum = sum(nums)
        lsum, ans = 0, 0

        for i in nums[:-1]:
            lsum += i 
            rsum -= i 
            if lsum >= rsum:
                ans += 1 
        
        return ans

File: 2358-number-of-ways-to-split-array/number-of-ways-to-split-array.cpp
class Solution {
public:
    int waysToSplitArray(vector<int>& nums) {
        long long rsum = accumulate(begin(nums), end(nums), 0LL), lsum = 0, ans = 0 ;
        for(int i = 0 ; i < nums.size() - 1 ; i ++){
            lsum += nums[i];
            rsum -= nums[i] ;
            if(lsum >= rsum)
                ans ++ ;
        }
        return ans;
    }
};